src/services/document-service.js

```diff
@@ -461,6 +474,67 @@ class DocumentService {
     }
   }
 
+  async regenerateDocument(documentKey) {
+    let lock;
+
+    try {
+      lock = await this.documentLockStore.takeLock(documentKey);
+
+      await this.transactionRunner.run(async session => {
+        const existingDocumentRevisions = await this._getAllDocumentRevisionsByKey(documentKey, session);
+
+        const document = this._buildDocumentFromRevisions(existingDocumentRevisions);
+
+        logger.info(`Saving document '${documentKey}' with revision ${document.revision}`);
+        await this.documentStore.save(document, { session });
+      });
+    } finally {
+      if (lock) {
+        await this.documentLockStore.releaseLock(lock);
+      }
+    }
+  }
+
+  async createRegenerateDocumentsBatch(user) {
```

Comment:
In order to have a sort of "entity" name, like "imports", and given also my suggestion [here](https://github.com/educandu/educandu/pull/142/files#r780988677), I think `createDocumentsRegenerationBatch` might be better.

Reply:
Renamed to: createDocumentRegenerationBatch as per Andraes' comment.